tests/cpu/fileformats/xmlutils/XMLReaderUtils_tests.cpp

```diff
@@ -45,13 +45,9 @@ OCIO_ADD_TEST(XMLReaderHelper, string_to_float_failure)
 
     const char str2[] = "12345";
     const size_t len2 = std::strlen(str2);
-    // All characters are parsed and this is more than the required length.
-    // The string to double function strtod does not stop at a given length,
-    // but we detect that strtod did read too many characters.
-    OCIO_CHECK_THROW_WHAT(OCIO::ParseNumber(str2, 0, len2 - 2, value),
-                          OCIO::Exception,
-                          "followed by unexpected characters");
-
+    // strtod with temporary storage will behave the same as from_chars
+    OCIO_CHECK_NO_THROW(OCIO::ParseNumber(str2, 0, len2 - 2, value));
```

Comment:
@hodoulp, this is the test that will detect the inconsistency. strtod-based from_chars will throw here without the duplication, whereas from_chars will succeed.

Reply:
Based on the only usage (read floats separated by a space, comma, etc. anyway the separator character is not a number) it's perhaps not a problem. Let me think about.

Reply:
To check, you could try making the string duplication conditional on `first + strlen(first) > last`, and see what codepaths trigger it.

Reply:
I did some checks with your pull request and the following code is working without the temporary string:
```
really_inline from_chars_result from_chars(const char * first, const char * last, double & value) noexcept
{
    errno = 0;
    if (!first || !last || first == last)
    {
        return {first, std::errc::invalid_argument};
    }

    char * endptr = nullptr;

    double
#ifdef _WIN32
        tempval = _strtod_l(first, &endptr, loc.local);
#else
        tempval = ::strtod_l(first, &endptr, loc.local);
#endif

    if (errno != 0)
    {
        return {first + (endptr - first), std::errc::result_out_of_range};
    }
    else if (endptr == first)
    {
        return {first, std::errc::invalid_argument};
    }
    else if (endptr <= last)
    {
        value = tempval;
        return {first + (endptr - first), {}};
    }
    else
    {
        return {first, std::errc::argument_out_of_domain};
    }
}
```

Reply:
The rest of the tests pass, so I'll comment the test suite (just in case it's needed in the future).